src/inference/src/dev/isync_infer_request.cpp

```diff
@@ -122,14 +128,28 @@ ov::ISyncInferRequest::FoundPort ov::ISyncInferRequest::find_port(const ov::Outp
                 node1->get_type_info() == node2->get_type_info() &&
                 node1->outputs().size() == node2->outputs().size() && node1->inputs().size() == node2->inputs().size());
     };
+    // Find port without caching work slow because we need each time iterate over all ports and compare different
+    // strings So use WA with caching in order to make 2+ calls for the same ports faster.
+    // Calculate hash for the port
+    size_t port_hash = ov::util::hash_combine(
+        std::vector<size_t>{std::hash<const ov::Node*>()(port.get_node()), std::hash<size_t>()(port.get_index())});
```

Comment:
In theory, cache does improve performance, but hash computation need cost time, if the hash computation is higher than  loop over all ports, it may cause performance drop. 
We should notice that most of models' input/output ports number does not exceed 10, it means that loop ports maybe not consume too much time, while hash computation will consume more than loop every ports, the performance maybe impact.
As I early tests, if we still run 102x find_ports(), the performance is worse than no caching(122 VS 160)

Actually the main point to impact performance is L298 and L302, which avoid 102 times find_port() calling and can get performance improved greatly for model with big number of input and output.

I have done another test:
This PR(Port cache + L298+L303): FPS=408
Only L298+L303 (No port cache): FPS=415

So I think except input/output number is very big, maybe exceed 100, other case port cache is not helpful.

Reply:
@riverlijunjie Are you really sure that diff 408 and 415 FPS is not related with infelicity between different runs?
I think hash calculation should be faster than strings comparison...

Reply:
At least it can prove that caching port doesn't help too much on performance, because after changing of L298 and L302, find_port() calling times has been decreased greatly.

I have done some tests and get below result for difference combination:
(Note, previous fps is got in debug version, and below is release version)

<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:dt="uuid:C2F41010-65B3-11d1-A29F-00AA00C14882"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=OneNote.File>
<meta name=Generator content="Microsoft OneNote 15">
</head>

<body lang=en-US style='font-family:Calibri;font-size:11.0pt'>
<!--StartFragment-->

<div style='direction:ltr'>

  | 1 | 2
-- | -- | --
No port cache | 351 | 361
Port cache | 373 | 357
No port cache   +  L298/L302 changes | 748 | 752
Port cache +  L298/L302 changes | 752 | 767
Master branch | 753 |  
</div>

<!--EndFragment-->
</body>

</html>


It seems that:
1) L298/L302 changes has achieved the similar fps with master branch
2) port caching does limited performance benefit, at least for the tested model.

Reply:
I also ran another model with 1 input and 1 output, to check the port caching benefit:

<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:dt="uuid:C2F41010-65B3-11d1-A29F-00AA00C14882"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=OneNote.File>
<meta name=Generator content="Microsoft OneNote 15">
</head>

<body lang=en-US style='font-family:Calibri;font-size:11.0pt'>
<!--StartFragment-->

<div style='direction:ltr'>



  | 1 | 2 | 3
-- | -- | -- | --
No port cache | 654 | 625 | 631
Port cache | 625 | 624 | 634



</div>

<!--EndFragment-->
</body>

</html>